library/xml/test/xml_test.rb

```diff
@@ -187,6 +187,50 @@
   end
 
   describe ".XMLToYCPString" do
+    context "regarding 'config:type' and 't' attributes:" do
+      it "recognizes the 't' attribute" do
+        input = "<test xmlns=\"http://www.suse.com/1.0/yast2ns\" xmlns:config=\"http://www.suse.com/1.0/configns\">\n" \
+          "  <foo t=\"symbol\">sym</foo>\n" \
+          "</test>\n"
+        expected = { "foo" => :sym }
+        expect(subject.XMLToYCPString(input)).to eq expected
+      end
+
+      it "recognizes the 'type' attribute (unnamespaced)" do
+        input = "<test xmlns=\"http://www.suse.com/1.0/yast2ns\" xmlns:config=\"http://www.suse.com/1.0/configns\">\n" \
+          "  <foo type=\"symbol\">sym</foo>\n" \
+          "</test>\n"
+        expected = { "foo" => :sym }
+        expect(subject.XMLToYCPString(input)).to eq expected
+      end
+
+      it "recognizes the 'config:type' attribute" do
+        input = "<test xmlns=\"http://www.suse.com/1.0/yast2ns\" xmlns:config=\"http://www.suse.com/1.0/configns\">\n" \
+          "  <foo config:type=\"symbol\">sym</foo>\n" \
+          "</test>\n"
+        expected = { "foo" => :sym }
+        expect(subject.XMLToYCPString(input)).to eq expected
+      end
+
+      it "recognizes the 'config:type' attribute even with a different prefix" do
+        input = "<test xmlns=\"http://www.suse.com/1.0/yast2ns\" xmlns:c=\"http://www.suse.com/1.0/configns\">\n" \
+          "  <foo c:type=\"symbol\">sym</foo>\n" \
+          "</test>\n"
+        expected = { "foo" => :sym }
+        expect(subject.XMLToYCPString(input)).to eq expected
+      end
+
+      it "in case of conflict, the shortest attribute wins (t over type over config:type)" do
```

Comment:
is it valid XML according to schema? I think it should forbid multiple attributes.

Reply:
OK, will fix to `raise`